braindecode/models/eegconformer.py

```diff
@@ -0,0 +1,313 @@
+    when the parameters were used first, e.g. n_filters_conv.
+
+    Parameters
+    ----------
+    - n_filters_conv: int
+        Length of kernels for the temporal convolution layer (first layer).
+    - n_filters_time: int
+        Number of temporal filters.
+    - filter_time_length: int
+        Length of the temporal filter.
+    - n_channels: int
+        Number of channels to be used as number of spatial filters.
+    - pool_time_length: int
+        Length of temporal poling filter.
+    - pool_time_stride: int
+        Length of stride between temporal pooling filters.
+    - drop_prob: float
+        Dropout rate of the convolutional layer.
+    - att_depth: int
+        Number of self-attention layers.
+    - att_heads: int
+        Number of attention heads.
+    - att_drop_prob: float
+        Dropout rate of the self-attention layer.
+    - final_fc_length: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes to predict (number of output filters of last layer).
+
+    References
+    ----------
+    .. [Song2022] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [ConformerCode] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+    """
+
+    def __init__(
+            self,
+            n_classes,
+            n_channels,
+            n_filters_conv=40,
+            n_filters_time=25,
+            filter_time_length=25,
+            n_kernel_avg_pool=75,
+            pool_time_stride=15,
+            drop_prob=0.5,
+            att_depth=6,
+            att_heads=10,
+            att_drop_prob=0.5,
+            final_fc_length=2440,
+    ):
+        super().__init__()
+        self.patch_embedding = PatchEmbedding(
+                n_filters_time=n_filters_time,
+                filter_time_length=filter_time_length,
+                n_channels=n_channels,
+                pool_time_length=n_kernel_avg_pool,
+                stride_avg_pool=pool_time_stride,
+                drop_prob=drop_prob)
+
+        self.transformer = _TransformerEncoder(
+            att_depth=att_depth,
+            emb_size=n_filters_conv,
+            att_heads=att_heads,
+            att_drop=att_drop_prob)
+
+        self.classification_head = ClassificationHead(
+            emb_size=n_filters_conv, final_fc_length=final_fc_length,
+            n_classes=n_classes)
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.patch_embedding(x)
+        x = self.transformer(x)
+        x = self.classification_head(x)
+        return x
+
+
+class PatchEmbedding(nn.Module):
```

Comment:
Shouldn't this also be private class? If it should be public we may add docs for arguments.

Reply:
I added, but I converted to private. Make more sense to used as private.